mbuild/utils/conversion.py

```diff
@@ -1,25 +1,137 @@
+
+    NOTE: fO IS TYPICALLY NOT IN THE OPLS DIHEDRAL EQUATION AND IS
+    ONLY USED TO TEST IF THIS FUNCTION CAN BE UTILIZED, AND
+    DETERMINE THE fO VALUE FOR LATER ENERGY SCALING IN MOLECULAR DYNAMICS
+    (MD) SIMULATIONS. THIS FUNCTION TESTS IF f0 IS ZERO (f0=0).
+
+    WARNING: The f0 term is the constant for the OPLS dihedral equation.
+    If the f0 term is not zero, the dihedral is not an exact conversion
+    from RB-torsions to an OPLS dihedral, which means the whole dihedral
+    potential energy is shifted by f0 the value.
     """
-    f1 = (-1.5 * c3) - (2 * c1)
-    f2 = c0 + c1 + c3
-    f3 = -0.5 * c3
-    f4 = -0.25 * c4
-    return np.array([f1, f2, f3, f4])
+    try:
+        error_tol = abs(float(error_tol))
+    except:
+        raise TypeError("ERROR: The error_tol variable must be a float.")
+
+    if not np.all(np.isclose(c5, 0, atol=error_tol, rtol=0)):
+        raise ValueError(
+            "ERROR: c5 must equal zero, so this conversion is not possible."
+        )
+
+    f0 = 2.0 * (c0 + c1 + c2 + c3 + c4 + c5)
+    if not np.all(np.isclose(f0, 0, atol=error_tol, rtol=0)):
+        warn(
+            "WARNING: f0 = 2 * (c0 + c1 + c2 + c3 + c4 + c5) is not zero. "
+            "The f0 term is the constant for the OPLS dihedral. "
+            "Since the f0 term is not zero, the dihedral is not an exact conversion "
+            "from RB-torsions to an OPLS dihedral, which means the whole dihedral "
+            "potential energy is shifted by f0 the value."
+        )
+
+    f1 = -2 * c1 - (3 * c3) / 2
+    f2 = -c2 - c4
+    f3 = -c3 / 2
+    f4 = -c4 / 4
+    return np.array([f0, f1, f2, f3, f4])
+
+
+def OPLS_to_RB(f0, f1, f2, f3, f4, error_tol=1e-4):
+    r"""Convert OPLS type to Ryckaert-Bellemans type dihedrals.
+
+    .. math::
+    OPLS_torsions &= f0/2 + f1/2*(1+cos(t)) + f2/2*(1-cos(2*t)) + \\
+                  &= f3/2*(1+cos(3*t)) + f4/2*(1-cos(4*t))
+
+    .. math::
+    RB_torsions &= c0 + c1*cos(psi) + c2*cos(psi)^2 + c3*cos(psi)^3 + \\
+                &= c4*cos(psi)^4 + c5*cos(5*psi)^5
+
+    where :math:`psi = t - pi = t - 180 degrees`
+
+    Parameters
+    ----------
+    f0, f1, f2, f3, f4 : OPLS dihedrals coeffs (in kcal/mol)
+    error_tol : float, default=1e-4
+        The acceptable absolute tolerance, which the f0 term can be from zero
+        without throwing a warning.
+
+    Returns
+    -------
+    RB_coeffs : np.array, shape=(6,)
+        Array containing the Ryckaert-Bellemans dihedrals
+        coeffs c0, c1, c2, c3, c4, and c5 (in kcal/mol)
+
+    Notes
+    -----
+    NOTE: fO IS TYPICALLY NOT IN THE OPLS DIHEDRAL EQUATION (i.e., f0=0).
+
+    WARNING: The f0 term is the constant for the OPLS dihedral equation,
+    which is converted and added to a constant for the RB torsions equation via the c0 coefficient.
+    If the f0 term is zero in the OPLS dihedral form or set to zero in this equation,
+    the dihedral is may not be an exact conversion, which means the whole dihedral
```

Comment:
minor grammar stuff. 

"the dihedral is may not be an exact conversion, which means the whole dihedra potential energy is shifted by real f0/2 the value.l"
something like:
"the dihedral is not an exact conversion; since this constant does not contribute to the force equation, this should provide matching results for MD,  but the energy for each diheral will be shifted by the 0/2 the value."

Reply:
Changed as requested